File: week-3/find-all-anagrams-in-a-string/find-all-anagrams-in-a-string.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        res = []
        if not s or not p or len(p) > len(s): return res
        p = sorted(p)
        
        for i in range(len(s) - len(p) + 1):
            if sorted(s[i:i + len(p)]) == p:
                res.append(i)
                
        return res

File: week-3/find-all-anagrams-in-a-string/find-all-anagrams-in-a-string.js
/**
 * @param {string} s
 * @param {string} p
 * @return {number[]}
 */
var findAnagrams = function(s, p) {
    const res = [];
    if (!s || !p || p.length > s.length) return res;
    
    p = p.split('').sort().join('');
    
    for (let i=0; i <= s.length - p.length; i++) {
        if (s.slice(i, i+p.length).split('').sort().join('') === p) res.push(i)
    }
    
    return res;
};
